--- a/arceos/modules/axhal/src/dummy.rs
+++ b/arceos/modules/axhal/src/dummy.rs
@@ -117,6 +117,8 @@
 impl IrqIf for DummyIrq {
     fn set_enable(_irq: usize, _enabled: bool) {}
 
+    fn set_priority(_irq: usize, _priority: u8) {}
+
     fn register(_irq: usize, _handler: IrqHandler) -> bool {
         false
     }
@@ -130,4 +132,18 @@
     }
 
     fn send_ipi(_irq: usize, _target: IpiTarget) {}
+
+    fn local_irq_save_and_disable() -> usize {
+        0
+    }
+
+    fn local_irq_restore(_flags: usize) {}
+
+    fn enable_irqs() {}
+
+    fn disable_irqs() {}
+
+    fn irqs_enabled() -> bool {
+        false
+    }
 }
